capivaraProjects/src/views/addUser.tsx

```diff
@@ -103,19 +116,57 @@ export class addUser extends React.Component<{},AddUserState>{
     private handleButtonPress = () =>{
 
         this.setState({ 
-                        phoneIsValid       : this.validate.phone(this.state.phoneInput),
-                        emailIsValid       : this.validate.email(this.state.emailInput),
-                        birthDateIsValid   : this.validate.dateBirth(this.state.birthDateInput),
-                        nameIsValid        : this.validate.name(this.state.nameInput),
-                        passwordIsValid    : this.validate.password(this.state.passwordInput),
-                        roleIsValid        : this.validate.name(this.state.roleInput), 
-                    });
+            phoneIsValid       : this.validate.phone(this.state.phoneInput),
+            emailIsValid       : this.validate.email(this.state.emailInput),
+            birthDateIsValid   : this.validate.dateBirth(this.state.birthDateInput),
+            nameIsValid        : this.validate.name(this.state.nameInput),
+            passwordIsValid    : this.validate.password(this.state.passwordInput),
+            roleIsValid        : this.validate.name(this.state.roleInput), 
+        });
+
+        if(this.checkInputState()){
```

Comment:
Me parece que você está repetindo as operações de validação, uma vez ali no setState, e outra dentro desse `checkInputState` 🤔 

Outra coisa é que não é muito saudável usar os valores do `this.state` logo depois de dar um `this.setState`. Se não me engano o React não garante se o valor do state vai estar atualizado ou não ainda.

Reply:
Sim, tive esse problema tive que fazer a checagem denovo. Mas pensei em uma solução